kv_store/src/bank/kv/application.rs

```diff
@@ -141,17 +142,26 @@ impl<DB: Database> ApplicationKVBank<DB> {
     }
 
     pub fn commit(&mut self) -> [u8; 32] {
-        let (cache, delete) = self.cache.take();
+        let (insert, delete) = self.cache.take();
 
         let mut persistent = self.persistent.write().expect(POISONED_LOCK);
 
-        cache
+        let cache = insert
             .into_iter()
             .filter(|(key, _)| !delete.contains(key))
-            .for_each(|(key, value)| persistent.set(key, value));
+            .map(|(key, value)| (key, Some(value)))
+            .collect::<Vec<_>>()
```

Comment:
Is there a reason to collect to vec then go back to an iter?

Reply:
Borrowing of `delete`.

Reply:
W8. It shouldn't make any sense 'cause I remove equal keys on set or delete